includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -248,11 +249,17 @@ protected function prepare_object_for_database( $request, $creating = false ) {
 
 		$variation->set_parent_id( absint( $request['product_id'] ) );
 
-		// Status.
+		// TODO Fix/Remove in future API version bump. This controls the wrong thing.
+		// https://github.com/woocommerce/woocommerce/issues/15215
 		if ( isset( $request['visible'] ) ) {
 			$variation->set_status( false === $request['visible'] ? 'private' : 'publish' );
 		}
 
+		// Status.
```

Comment:
@justinshreve An update to the WordPress.com API rolls out for all clients at once.

And update to the WooCommerce API means multiple versions can be active at the same time since it's self hosted.

Therefore versions need to be increased in our case if the behavior changes.